torchaudio/functional/functional.py

```diff
@@ -248,7 +251,13 @@ def amplitude_to_DB(
     x_db -= multiplier * db_multiplier
 
     if top_db is not None:
-        x_db = x_db.clamp(min=x_db.max().item() - top_db)
+        dims = x_db.dim()
+        if dims > 3:
```

Comment:
Here's an [example](https://github.com/pytorch/audio/blob/master/torchaudio/transforms.py#L256) on how to pack and unpack a tensor from (..., channel, freq, time) to (batch, channel, freq, time) and back again to (..., channel, freq, time) in a torchscriptable manner.